CMakeLists.txt

```diff
@@ -350,9 +347,21 @@ if (ENABLE_EXTRA_WARNINGS)
     -Wpointer-arith
     -Wpointer-sign
     -Wmissing-format-attribute
-    -Wsuggest-attribute=noreturn
     -Wold-style-definition
     -Wundef)
+
+if ("${CMAKE_C_COMPILER_ID}" MATCHES "Clang")
```

Comment:
We might want to add the same to `configure.ac` with:
```
if test "x$ac_compiler_gnu" = "xyes"; then
...
```

Reply:
clang is kinda a gnu compiler, at least it pretends to be one. this condition won't help here.